Mail/Views/Menu Drawer/Folders/FoldersListView.swift

```diff
@@ -18,27 +18,75 @@
 
 import InfomaniakCore
 import MailCore
+import MailResources
 import RealmSwift
+import SwiftModalPresentation
 import SwiftUI
 
 struct FoldersListView: View {
     @EnvironmentObject private var mainViewState: MainViewState
+    @EnvironmentObject private var mailboxManager: MailboxManager
+
+    @ModalState private var isShowingCreateFolderAlert = false
+    @State private var currentFolder: Folder?
 
     private let folders: [NestableFolder]
     private let hasSubFolders: Bool
+    private let isUserFoldersList: Bool
```

Comment:
Do not use `isUserFoldersList` but check for `folder.role`. If folder has a role do not allow rename.